test/fluree/db/transact/stable_context_test.clj

```diff
@@ -0,0 +1,91 @@
+(ns fluree.db.transact.stable-context-test
+  (:require [clojure.test :refer :all]
+            [fluree.db.dbproto :as dbproto]
+            [fluree.db.test-utils :as test-utils]
+            [fluree.db.json-ld.api :as fluree]
+            [test-with-files.tools :refer [with-tmp-dir] :as twf]))
+
+
+;; ensures default context remains stable across many commits.
+
+(deftest ^:integration default-context-stability-memory
+  (let [conn     (test-utils/create-conn)
+        ledger   @(fluree/create conn "ctx/stability-mem" {:defaultContext {:id   "@id"
+                                                                        :type "@type"
+                                                                        :ex   "http://example.org/ns/"
+                                                                        :blah "http://blah.me/wow/ns/"}})
+        db1      @(test-utils/transact ledger [{:id      :blah/one
+                                                :ex/name "One"}])
+        db1-load (fluree/db @(fluree/load conn "ctx/stability-mem"))
+        db2      @(test-utils/transact ledger [{:id      :blah/two
+                                                :ex/name "Two"}])
+        db2-load (fluree/db @(fluree/load conn "ctx/stability-mem"))
+        db3      @(test-utils/transact ledger [{:id      :blah/three
+                                                :ex/name "Three"}])
+        db3-load (fluree/db @(fluree/load conn "ctx/stability-mem"))]
+
+    (testing "Loaded default context is same as initial db's"
+      (is (= (dbproto/-default-context db1-load)
+             (dbproto/-default-context db1))))
+
+    (testing "Second transaction default context is same as initial db's"
+      (is (= (dbproto/-default-context db2-load)
+             (dbproto/-default-context db1))))
+
+    (testing "Third transaction default context is same as initial db's"
+      (is (= (dbproto/-default-context db3-load)
+             (dbproto/-default-context db1))))
+
+    (testing "Query after the 3rd load is using original default context"
+      (is (= [{:id      :blah/three
+               :ex/name "Three"}
+              {:id      :blah/two
+               :ex/name "Two"}
+              {:id      :blah/one
+               :ex/name "One"}]
+             @(fluree/query db3-load '{:select {?s [:*]}
+                                       :where  [[?s :ex/name nil]]}))))))
+
+
+(deftest ^:integration default-context-stability-file
+  (with-tmp-dir storage-path
+                (let [conn     @(fluree/connect
```

Comment:
Small thing but it would be good to configure our editors to treat `with-tmp-dir` as a two-level indent form instead of an align-args form. So you would have this instead:

```clojure
(with-tmp-dir storage-path
  (let ...))
```

Reply:
In Cursive you can either do that by configuring the indent directly or by telling it to resolve this as one of the core `with-...` forms it already knows about from the "context actions" menu (right click on `with-tmp-dir` and select "Show Context Actions"). One of those might get it to understand that you're defining `storage-path` here as well.

Reply:
Fixed!